bolts/transforms/tabular.py

```diff
@@ -0,0 +1,178 @@
+        super().__init__(inplace=inplace)
+        if not (0 <= q_min <= 1):
+            raise ValueError("q_min must be in the range [0, 1]")
+        if not (0 <= q_max <= 1):
+            raise ValueError("q_max must be in the range [0, 1]")
+        if q_min > q_max:
+            raise ValueError("'q_min' cannot be greater than 'q_max'.")
+        self.q_min = q_min
+        self.q_max = q_max
+
+    @staticmethod
+    def _compute_quantile(q: float, sorted_values: Tensor) -> Tensor:
+        q_ind_frac, q_ind_int = math.modf(q * (len(sorted_values) - 1))
+        q_ind_int = int(q_ind_int)
+        q_quantile = sorted_values[q_ind_int]
+        if q_ind_frac > 0:
+            q_quantile += (sorted_values[q_ind_int + 1] - q_quantile) * q_ind_frac
+        return q_quantile
+
+    @implements(TabularTransform)
+    def _fit(self, data: Tensor) -> None:
+        sorted_values = data.sort(dim=0, descending=False).values
+        # Compute the 'lower quantile'
+        q_min_quantile = self._compute_quantile(q=self.q_min, sorted_values=sorted_values)
+        # Compute the 'upper quantile'
+        q_max_quantile = self._compute_quantile(q=self.q_max, sorted_values=sorted_values)
+        # Compute the interquantile range
+        self.iqr = q_max_quantile - q_min_quantile
+        self.median = self._compute_quantile(q=0.5, sorted_values=sorted_values)
+
+    @implements(TabularTransform)
+    def _inverse_transform(self, data: Tensor) -> None:
+        data *= self.iqr
+        data += self.median
+
+    @implements(TabularTransform)
+    def _transform(self, data: Tensor) -> None:
+        data -= self.median
+        data /= self.iqr.clamp_min(self._EPS)
+
+
+class MinMaxNormalization(TabularTransform):
+
+    orig_max: Tensor
+    orig_min: Tensor
+    orig_range: Tensor
+
+    def __init__(self, new_min: float = 0.0, new_max: float = 1.0, inplace: bool = False) -> None:
+        super().__init__(inplace=inplace)
+        if new_min > new_max:
+            raise ValueError("'new_min' cannot be greater than 'new_max'.")
+        self.new_min = new_min
+        self.new_max = new_max
+        self.new_range = self.new_max - self.new_min
+
+    @implements(TabularTransform)
+    def _fit(self, data: Tensor) -> None:
+        self.orig_min = torch.min(data, dim=0, keepdim=True).values
+        self.orig_max = torch.max(data, dim=0, keepdim=True).values
+        self.orig_range = self.orig_max - self.orig_min
+
+    @implements(TabularTransform)
+    def _inverse_transform(self, data: Tensor) -> None:
+        data -= self.new_min
+        data /= self.new_range + self._EPS
+        data *= self.orig_range
+        data += self.orig_min
+
+    @implements(TabularTransform)
+    def _transform(self, data: Tensor) -> None:
+        data -= self.orig_min
+        data /= self.orig_range.clamp_min(self._EPS)
+        data *= self.new_range
+        data += self.new_min
```

Comment:
I know this is an inplace operation, but this doesn't feel good

Reply:
I'd much prefer if `data` was returned

Reply:
How would this work with the `col_indexes`, the specification of which would lead to a subset of the data being returned?

Reply:
will `data[:, self.col_indexes] = self._transform(data[:, self.col_indexes])` work?

Reply:
I suppose it would though I'd rather keep it as it is to enforce inplaceness and thus avoid unnecessary copies of the data after the clone.

Reply:
in that case, could the operations be made to be the explicit inplace ones e.g.
```
data.sub_(self.orig_min)
data.div_(self.orig_range.clamp_min(self._EPS))
etc
```